### PR TITLE
[doc] fix missing Message implementation for ResponseActFuture

### PR DESCRIPTION
**Pull request by Freyskeyd**
In the documentation the `ResponseActFuture` example is missing an important part that make it unclear for newcomer.

Signed-off-by: Freyskeyd <simon.paitrault@gmail.com>

### CODE REVIEW

**Comment by JohnTitor** [comment]
Yea, this should be added to the example but I'm not sure what a complete example is. Are there any other things that we should add?

**Comment by Freyskeyd** [comment]
here you're, this is a lot of boilerplate for making this example compile but I don't know how to simplify more. :)

**Comment by JohnTitor** [comment]
Oh, thanks! It seems a good example. We can leave this complete example (and hide a part of it) to run a document test, or leave the ignore flag and exclude some code. I think this example is too long to document as-is.

I'd like to hear other opinions. cc: @actix/contributors

**Comment by Freyskeyd** [comment]
Yes, I also think it's a lot of code to demonstrate how to handle a `ResponseActFuture`.

Hidding all the common things like actor's implementation and unimportant message.

But keeping `type`s and `DeferredWork`'s message implementation is the crucial part.

**Comment by JohnTitor** [comment]
I thought about this change for a while. In conclusion, it's a good idea to leave a perfect example in the code and show a short example in the documentation. 
@Freyskeyd Could you comment out unimportant parts like this?

```rust
/// important part
/// # unimportant part, comment out in documentation
```

**Comment by Freyskeyd** [comment]
@JohnTitor is it ok for you that way?

File: src/fut/mod.rs [hunk 1]
```diff
@@ -52,35 +52,77 @@ use crate::actor::Actor;
...
+/// # #[derive(Message)]
+/// # struct OtherMessage{}
+/// 
 /// impl Handler<DeferredWork> for OriginalActor {
 ///     // Notice the `Response` is an `ActorFuture`-ized version of `Self::Message::Result`.
-///     type Result = ResponseActFuture<Self, (), Error>;
-///
-///     fn handle(&mut self, _msg: Refresh, ctx: &mut Context<Self>) -> Self::Result {
+///     type Result = ResponseActFuture<Self, OriginalActorResponse, MessageError>;
+///
```
Comment (JohnTitor) [inline src/fut/mod.rs]: There is a trailing space, should be removed.
```suggestion
///
```

File: src/fut/mod.rs [hunk 2]
```diff
@@ -52,35 +52,77 @@ use crate::actor::Actor;
...
 ///         // this creates a `Future` representing the `.send` and subsequent `Result` from
 ///         // `other_actor`
-///         let send_to_other = self.other_actor_addr
-///             .send(OtherMessage::new())
-///             .map_err(Error::from);
-///
+///         let send_to_other = self.other_actor
+///             .send(OtherMessage {})
+///             .map_err(|_error| ());
+///
```
Comment (JohnTitor) [inline src/fut/mod.rs]: same here
```suggestion
///
```

File: src/fut/mod.rs [hunk 3]
```diff
@@ -52,35 +52,77 @@ use crate::actor::Actor;
...
-///
+///         let send_to_other = self.other_actor
+///             .send(OtherMessage {})
+///             .map_err(|_error| ());
+/// 
 ///         // Wrap that `Future` so subsequent chained handlers can access
 ///         // the `actor` (`self` in the  synchronous code) as well as the context.
 ///         let send_to_other = actix::fut::wrap_future::<_, Self>(send_to_other);
-///
+///
```
Comment (JohnTitor) [inline src/fut/mod.rs]:
```suggestion
///
```

File: src/fut/mod.rs [hunk 4]
```diff
@@ -52,35 +52,77 @@ use crate::actor::Actor;
...
 ///         let send_to_other = actix::fut::wrap_future::<_, Self>(send_to_other);
-///
+/// 
 ///         // once the wrapped future resolves, update this actor's state
 ///         let update_self = send_to_other.map(|result, actor, _ctx| {
 ///             // Actor's state updated here
 ///             actor.inner_state.update_from(result);
 ///         });
-///
+///
```
Comment (JohnTitor) [inline src/fut/mod.rs]:
```suggestion
///
```

**Comment by Freyskeyd** [comment]
> LGTM with some nits

fixed :)

**Comment by JohnTitor** [comment]
Thanks!

**Pull request merged**